File: MyFirstProject/luhn.py
#The pattern follows 2,1,2,1...
#I should check if it is bigger than 10 or not.

creditnum = "6500830926349532"
multiply = 1
sum = 0
temp = 0

for i in creditnum:
    if multiply == 1:
        multiply = 2
    else:
        multiply = 1

    temp = int(i) * multiply
    

    if temp > 9:
        temp = (temp % 10) + int(temp / 10)
    
    sum += temp

if sum % 10 == 0:
    print("The cart is valid")
else:
    print("The cart is not valid")

File: MyFirstProject/luhn.cpp
#include <iostream>

int main(){

    std::string x = "6500830926349533";
    int temp = 0;
    int sum = 0;

    for (int i = 0; i < x.length(); i++)
    {
        if (i % 2 == 0)
        {
            temp = 2 * (x.at(i) - '0');
        }
        else
        {
            temp = x.at(i) - '0';
        }

        if (temp > 9)
        {
            temp = (temp % 10) + (temp / 10);
        }

        sum += temp;
    }

    if (sum % 10 == 0)
    {
        std::cout << "The card is valid" << std::endl;
    }
    else
    {
        std::cout << "The card is not valid" << std::endl;
    }
    
    return 0;
}